src/idl/src/annotation.c

```diff
@@ -38,17 +38,18 @@ annotate_id(
 
   if (idl_mask(node) & IDL_MEMBER) {
     idl_member_t *member = (idl_member_t *)node;
+    assert(member->declarators);
     if (idl_next(member->declarators)) {
       idl_error(pstate, idl_location(annotation_appl),
         "@id cannot be applied to members with multiple declarators");
       return IDL_RETCODE_SEMANTIC_ERROR;
-    } else if (member->id.annotation) {
+    } else if (member->declarators->id.annotation) {
       idl_error(pstate, idl_location(annotation_appl),
         "@id conflicts with earlier annotation");
       return IDL_RETCODE_SEMANTIC_ERROR;
     }
-    member->id.annotation = annotation_appl;
-    member->id.value = literal->value.uint32;
+    member->declarators->id.annotation = annotation_appl;
+    member->declarators->id.value = (literal->value.uint32 & 0x0FFFFFFFu);
```

Comment:
Why the `0x0FFFFFFFu`? Ensuring it's within range?

Reply:
This is to generate the rollover of the member-id values, as the consumers of the member ids (the xtypes implementations) only use the 28 least significant bits.